pkg/plugin/aggregation/run.go

```diff
@@ -212,6 +219,27 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	}
 }
 
+func getOrderedPlugins(plugins []plugin.Interface) [][]plugin.Interface {
```

Comment:
Since I really want to emphasize readability and maintainability and optimizations are not that important we should really consider simply using the `sort` package to sort these objects in terms of `order` then `name`; once they are ordered you can loop through and group them trivially.

That may be the most succinct way to execute this and the number of plugins will always (unless trying to prove a point) be < 100.

Open to other approaches but just in case you thought I wouldn't like using that simple approach I wanted to let you know I think its an easy, clear, maintainable option.